File: backend/app/schemas/complaint.py
from __future__ import annotations

from pydantic import BaseModel, Field, EmailStr


class ComplaintCreate(BaseModel):
    complaint_text: str = Field(min_length=5, max_length=8000)
    location_text: str | None = Field(default=None, max_length=512)
    image_url: str | None = Field(default=None, max_length=2048)
    email: EmailStr | None = Field(default=None, description="Email for public users to track their complaint")


class ComplaintStatus(BaseModel):
    complaint_id: str
    status: str
    current_step: str | None = None
    category: str | None = None
    agency: str | None = None
    confidence: float | None = None
    work_order_id: str | None = None
    priority: str | None = None
    citizen_email_preview: str | None = None
    email: str | None = None
    complaint_text: str | None = None
    location_text: str | None = None
    created_at: str | None = None
    updated_at: str | None = None

File: frontend/types/complaint.ts
export type ComplaintCreate = {
  complaint_text: string;
  location_text?: string | null;
  image_url?: string | null;
  email?: string | null;
};

export type ComplaintStatus = {
  complaint_id: string;
  status: string;
  current_step?: string | null;
  category?: string | null;
  agency?: string | null;
  confidence?: number | null;
  work_order_id?: string | null;
  priority?: string | null;
  citizen_email_preview?: string | null;
  email?: string | null;
  complaint_text?: string | null;
  location_text?: string | null;
  created_at?: string | null;
  updated_at?: string | null;
};

